src/client-side-metrics/gcp-metrics-handler.ts

```diff
@@ -28,6 +28,12 @@ const {
   PeriodicExportingMetricReader,
 } = require('@opentelemetry/sdk-metrics');
 
+// The variable below is the singleton map from projects to instrument stacks
+// which exists so that we only create one instrument stack per project. This
+// will eliminate errors due to the maximum sampling period.
+const defaultProjectToInstruments: {[projectId: string]: MetricsInstruments} =
```

Comment:
How does this fix the issue? Won't there still be an issue if multiple projects write at the same time?